base/applications/kbswitch/kbswitch.c

```diff
@@ -147,35 +147,43 @@ static HICON
 CreateTrayIcon(LPTSTR szLCID)
 {
     LANGID LangID;
-    TCHAR szBuf[3];
+    TCHAR szBuf[4];
     HDC hdc;
     HBITMAP hbmColor, hbmMono, hBmpOld;
     RECT rect;
     HFONT hFontOld, hFont;
     ICONINFO IconInfo;
     HICON hIcon;
     LOGFONT lf;
+    BITMAPINFO bmi;
 
     /* Getting "EN", "FR", etc. from English, French, ... */
-    LangID = (LANGID)_tcstoul(szLCID, NULL, 16);
-    if (!GetLocaleInfo(LangID, LOCALE_SISO639LANGNAME, szBuf, ARRAYSIZE(szBuf)))
+    LangID = LOWORD(_tcstoul(szLCID, NULL, 16));
+    if (!GetLocaleInfo(LangID, LOCALE_SABBREVLANGNAME | LOCALE_NOUSEROVERRIDE,
+                       szBuf, ARRAYSIZE(szBuf)))
     {
         StringCchCopy(szBuf, ARRAYSIZE(szBuf), _T("??"));
     }
-    CharUpper(szBuf);
+    szBuf[2] = 0; /* "ENG" --> "EN" etc. */
+
+    /* Prepare for DIB (device-independent bitmap) */
+    ZeroMemory(&bmi, sizeof(bmi));
+    bmi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bmi.bmiHeader.biWidth = CX_ICON;
+    bmi.bmiHeader.biHeight = CY_ICON;
+    bmi.bmiHeader.biPlanes = 1;
+    bmi.bmiHeader.biBitCount = 24;
 
     /* Create hdc, hbmColor and hbmMono */
     hdc = CreateCompatibleDC(NULL);
-    hbmColor = CreateCompatibleBitmap(hdc, CX_ICON, CY_ICON);
+    hbmColor = CreateDIBSection(hdc, &bmi, DIB_RGB_COLORS, NULL, NULL, 0);
```

Comment:
The 24 bpp DIB is a device-independent full color bitmap. So it doesn't need the color table if conversion was correct.

Reply:
The device-dependent bitmap (DDB) has a fear that conversion might be failed if the device was changed.

Reply:
![image](https://user-images.githubusercontent.com/2107452/192408861-b27a7ced-bfaa-49fc-b13b-a57bcc8fe25b.png)
Looks like monochrome.

Reply:
I guess that if the CreateCompatibleDC call was called on the hdc of the traybar's window, it could have been in color.
But why I'm overall surprised why this code would need any change, is I know that so far, the kbswitch was drawing a perfectly colored icon in the traybar, so why is it that sometimes it would draw a monochrome-looking one instead? :thinking: 
See for example:
https://i.ytimg.com/vi/Ne88Is2cymQ/maxresdefault.jpg
![image](https://user-images.githubusercontent.com/1969829/192550629-b0827f6c-d7f5-4c17-a8b3-61ce694731d7.png)

And switching to french didn't cause problem. But why for japanese (at least in your new tests) the "JA" icon now appears in black with the existing code, is a mystery for me.

Reply:
That bitmap is passed to `CreateIconIndirect` function. That is non-compatible.

Reply:
DDB was not full color. Rendering Japanese text used antialiasing. Antialiasing used gray colors. The DDB lacks palette. Darkblue color is ignored.

Reply:
Maybe our palette optimization was poor.

Reply:
If you isolate this "CreateTrayIcon" (the one before the fix from this PR) into a standalone test-program that just creates a tray icon (or any other kind of window on which the icon can be drawn), that you then run on Windows (japanese or not): would the created bitmap/icon be in color or not? (I would suggest testing that on Windows XP/2k3 before looking at e.g. Windows 10)

Reply:
I made a test program: https://jira.reactos.org/secure/attachment/62796/DDBTest.zip

ReactOS:
![DDBTest-ReactOS](https://user-images.githubusercontent.com/2107452/192652590-17c672d0-0d29-4955-99ef-abde910afe42.png)

WinXP:
![DDBTest-WinXP](https://user-images.githubusercontent.com/2107452/192652612-5737dd6c-0037-4d6c-8b4d-dae11d7d6a36.png)

Win2k3:
![DDBTest-Win2k3](https://user-images.githubusercontent.com/2107452/192652379-a48adbcf-3dd4-4f7f-8a71-a676b64a35df.png)

Win10:
![DDBTest-Win10](https://user-images.githubusercontent.com/2107452/192652611-ce038f55-9c33-4484-90c4-9095ee66e104.png)